nucliadb_vectors/src/data_point_provider/merge_worker.rs

```diff
@@ -23,19 +23,20 @@ use nucliadb_core::fs_state;
 use tracing::*;
 
 use super::merger::{MergeQuery, MergeRequest};
-use super::{State, VectorR};
+use super::work_flag::WorkFlag;
+use super::State;
 use crate::data_point::{DataPoint, DpId};
+use crate::VectorR;
 
-pub struct Worker(PathBuf);
+pub struct Worker(PathBuf, WorkFlag);
```

Comment:
It could be create to turn that tuple struct to classic one and naming its fields for clarity purpose.

Reply:
Done at [83d676f](https://github.com/nuclia/nucliadb/pull/563/commits/83d676fa207252d6791cc50184c5914ca8e4adbc)